deploy/packages/emqx.service

```diff
@@ -5,13 +5,34 @@ After=network.target
 [Service]
 User=emqx
 Group=emqx
-Type=forking
+
+# The ExecStart= is foreground, so 'simple' here
+Type=simple
 Environment=HOME=/var/lib/emqx
-ExecStart=/usr/bin/emqx start
+
+# Enable logging to file
+Environment=EMQX_LOG__TO=default
+
+# Start 'foregroun' but not 'start' (daemon) mode.
+# Because systemd monitor/restarts 'simple' services
+ExecStart=/usr/bin/emqx foreground
+
+# Give EMQX enough file descriptors
 LimitNOFILE=1048576
-ExecStop=/usr/bin/emqx stop
+
+# ExecStop is commented out so systemd will send a SIGTERM when 'systemctl stop'.
+# SIGTERM is handled by EMQX and it then performs a graceful shutdown
+# It's better than command 'emqx stop' because it needs to ping the node
```

Comment:
For my own information: `emqx stop` is worse because of the ping, and that is bad because the node might be unresponsive?

Reply:
yes, the ping relies on Erlang distribution being responsive (and the remove node is able to connect with a correct cookie).
if a user deletes the generated configs, and changed cookie in emqx.conf, the only way to stop is to `kill -9`  (similar things happened before -- a user changed node name in the config before stopping the old)